components/settings/settings-form.tsx

```diff
@@ -6,6 +6,7 @@ import Image from "next/image";
 import { useForm, useWatch } from "react-hook-form";
 import { zodResolver } from "@hookform/resolvers/zod";
 import { useAccount, useEnsAvatar, useEnsName } from "wagmi";
+import isURL from "validator/lib/isURL";
```

Comment:
I was just using anything that works. Happy to do it using zod :+1:

Reply:
I was playing around with this some more and zod's url validation is a bit more permissive than I'd like. I can change the code so that it only tries to generate the preview when the user de-focuses the input field, which drastically improves the situation, but then, even entering a broken url like `http//example.org/image.png` (see the missing colon) crashes the app which is bad. `isURL` is a reliable option here that allows us to keep the rendering while the user types and the moment they produce a valid URL the image pops in. 

We could of course roll our own url validation with just a few lines of code. Let me know if you prefer that instead @bitbeckers.

Reply:
Great, this is fine. Just wanted to make sure the step to not use zod had a reason